### DISCUSSION TITLE
Do we have an option to opt out of randomized classnames in production?

### DISCUSSION THREAD

**Discussion by ZenaMel** [facebook/docusaurus#10009] [Q&A]
When building the site, the classnames for native Docusaurus components receive randomized postfixes e.g.:

In build:
![image](https://github.com/facebook/docusaurus/assets/47085752/b1e9ae5c-2b7a-417b-8213-0ab08643e086)

In dev:
![image](https://github.com/facebook/docusaurus/assets/47085752/fab5e0fd-0969-4971-ae84-f071aba1b1b7)

Is there an option we can use to opt out of this optimization step? I'm aware of potential performance drawbacks.

Motivation: This would simplify adding custom CSS on native components that are not targetable via the exposed classnames found [here](https://docusaurus.io/docs/styling-layout#theme-class-names)

**Comment by slorber** [comment]
This "hash" is there for a reason, to automatically avoid conflicts between similar class names we (+ third party) theme authors might use. For example `.container` is a quite common class name that could be used by multiple persons, but `.container_abc` and `.container_def` are not. We don't provide that option on purpose, but you could use `configureWebpack` and find a way to do so (not sure it's a great idea though)

In the document linked, we mention you can create CSS selectors that ignore the hash suffix:

> CSS module class names. These class names have a hash in production (codeBlockContainer_RIuc) and are appended with a long file path in development. They are considered implementation details and you should almost always avoid targeting them in your custom CSS. If you must, you can use an [attribute selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors) ([class*='codeBlockContainer']) that ignores the hash.

Note that you can also swizzle wrap components and add your own stable class names if we don't provide any.

And if you detail the CSS targeting use case you have, we could consider adding new stable classes to help you target things more reliably.

**Reply by ZenaMel** [reply]
Our use-case was the Details element. We went with this targeting method which works fine in both contexts:

```scss
/* STYLING FOR COLLAPSIBLES (<details> element) */
details.alert--info {
.......................................
    > div > div {
.......................................    
    }
}
```
